CMake build fails on Termux due to hardcoded installation paths

The current `CMakeLists.txt` uses hardcoded absolute paths for installing headers and source files.

```cmake
install(DIRECTORY ./include/ DESTINATION /usr/include/)
install(DIRECTORY ./c/ DESTINATION /usr/src/${PROJECT_NAME}/)
```

This causes the build to fail on systems like Termux where `/usr/include` does not exist or is not the correct installation prefix. The error is:

```
CMake Error at cmake_install.cmake:73 (file):
  file INSTALL cannot make directory "/usr/include/": No such file or
  directory.
```

This can be fixed by replacing the hardcoded paths with relative paths, which respect the `CMAKE_INSTALL_PREFIX` variable.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,5 +132,5 @@ if (SIMSIMD_BUILD_SHARED)
     )
 endif ()
 
-install(DIRECTORY ./include/ DESTINATION /usr/include/)
-install(DIRECTORY ./c/ DESTINATION /usr/src/${PROJECT_NAME}/)
+install(DIRECTORY include/ DESTINATION include)
+install(DIRECTORY c/ DESTINATION share/doc/${PROJECT_NAME}/src)

